### PR TITLE
TSK-1666: Cheap Fix

### PR DESCRIPTION
**Pull request by Tristan2357**
https://sonarcloud.io/dashboard?branch=TSK-1666&id=Tristan2357_taskana

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
Indeed this is a cheap fix. :D

I am willing to accept it for now. But only if we create a proper Test which verifies that every implementation of `QueryParameter` uses only fields with @JsonProperty.

**Review by holgerhagen — DISMISSED** [review]

File: rest/taskana-rest-spring/pom.xml [hunk 1]
```diff
@@ -164,6 +164,12 @@
       <artifactId>spring-boot-starter-validation</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.tngtech.archunit</groupId>
+      <artifactId>archunit</artifactId>
+      <version>0.20.1</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/pom.xml:172]:
```suggestion
    <dependency>
      <groupId>com.tngtech.archunit</groupId>
      <artifactId>archunit-junit5</artifactId>
      <version>${version.archunit}</version>
      <scope>test</scope>
    </dependency>

```

Comment (mustaphazorgati) [reply rest/taskana-rest-spring/pom.xml:172]: Please replace `archunit-junit5-api` and `archunit-junit5-engine` dependency in taskana-core/pom.xml with this aggregator dependency. Thanks!

File: rest/taskana-rest-spring/src/test/java/pro/taskana/SpringArchitectureTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.jupiter.api.Test;
+
+public class SpringArchitectureTest {
+  private static final List<String> TASKANA_SUB_PACKAGES =
+      List.of(
+          "pro.taskana.common.rest",
+          "pro.taskana.classification.rest",
+          "pro.taskana.task.rest",
+          "pro.taskana.workbasket.rest",
+          "pro.taskana.monitor.rest");
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/SpringArchitectureTest.java:28]: not used

File: rest/taskana-rest-spring/src/test/java/pro/taskana/SpringArchitectureTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    // time intensive operation should only be done once
+    importedClasses = new ClassFileImporter().importPackages("pro.taskana", "acceptance");
+  }
+
+  @Test
+  void should_AnnotateAllFieldsWithJsonProperty_When_ImplementingQueryParameter() {
+    ArchRule myRule =
+        classes()
+            .that()
+            .implement(pro.taskana.common.rest.QueryParameter.class)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/SpringArchitectureTest.java:42]: import instead of qualify please

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Comment by mustaphazorgati** [comment]
Please rebase to fix CI issues

File: rest/taskana-rest-spring/pom.xml (hunk 1 above)
Comment (Tristan2357) [reply rest/taskana-rest-spring/pom.xml:172]: according to the module guide of archunit, this module does not exist

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**